### PR TITLE
Add oc_objectvalidator to upgrade check

### PR DESCRIPTION
**Pull request by sdodson**
We need to add failure conditions and references to documentation for how to remedy situations.

### CODE REVIEW

**Comment by sdodson** [comment]
@enj ok, so at least for the namespace bits I've verified that this aborts the upgrade. Where can we point people for documentation on how to fix the situation?

**Comment by sdodson** [comment]
```
PLAY [Check 3.5 items] *********************************************************

TASK [setup] *******************************************************************
ok: [ose3-master.example.com]

TASK [Verify sdn data is safe to upgrade] **************************************
fatal: [ose3-master.example.com]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "results": {
        "namespaces that use reserved names and were not created by infrastructure components": [
            {
                "apiVersion": "v1", 
                "kind": "Namespace", 
                "metadata": {
                    "annotations": {
                        "openshift.io/description": "", 
                        "openshift.io/display-name": "", 
                        "openshift.io/requester": "system:admin", 
                        "openshift.io/sa.scc.mcs": "s0:c8,c2", 
                        "openshift.io/sa.scc.supplemental-groups": "1000060000/10000", 
                        "openshift.io/sa.scc.uid-range": "1000060000/10000"
                    }, 
                    "creationTimestamp": "2017-03-06T19:59:12Z", 
                    "name": "kube-lolhax", 
                    "namespace": "", 
                    "resourceVersion": "17848", 
                    "selfLink": "/api/v1/namespaceskube-lolhax", 
                    "uid": "5e87a3d4-02a7-11e7-b158-525400ec778b"
                }, 
                "spec": {
                    "finalizers": [
                        "openshift.io/origin", 
                        "kubernetes"
                    ]
                }, 
                "status": {
                    "phase": "Active"
                }
            }
        ]
    }, 
    "state": "list"
}

MSG:

All objects are not valid.
```

**Comment by sdodson** [comment]
@tbielawa you should just add your petset validator to the validator.yml playbook as a task i guess

**Comment by sdodson** [comment]
Current output.
```
PLAY [Verify 3.5 specific upgrade cehcks] **************************************

TASK [setup] *******************************************************************
ok: [ose3-master.example.com]

TASK [Check for invalid namespaces and SDN errors] *****************************
fatal: [ose3-master.example.com]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "results": {
        "namespaces that use reserved names and were not created by infrastructure components": [
            {
                "apiVersion": "v1", 
                "kind": "Namespace", 
                "metadata": {
                    "annotations": {
                        "openshift.io/description": "", 
                        "openshift.io/display-name": "", 
                        "openshift.io/requester": "system:admin", 
                        "openshift.io/sa.scc.mcs": "s0:c8,c2", 
                        "openshift.io/sa.scc.supplemental-groups": "1000060000/10000", 
                        "openshift.io/sa.scc.uid-range": "1000060000/10000"
                    }, 
                    "creationTimestamp": "2017-03-06T19:59:12Z", 
                    "name": "kube-lolhax", 
                    "namespace": "", 
                    "resourceVersion": "17848", 
                    "selfLink": "/api/v1/namespaceskube-lolhax", 
                    "uid": "5e87a3d4-02a7-11e7-b158-525400ec778b"
                }, 
                "spec": {
                    "finalizers": [
                        "openshift.io/origin", 
                        "kubernetes"
                    ]
                }, 
                "status": {
                    "phase": "Active"
                }
            }
        ]
    }, 
    "state": "list"
}

MSG:

All objects are not valid.   If you are a supported customer please contact Red Hat Support providing the complete output above. If you are not a customer please contact aos-devel@redhat.com for assistance.


PLAY RECAP *********************************************************************
localhost                  : ok=11   changed=0    unreachable=0    failed=0   
ose3-master.example.com    : ok=131  changed=9    unreachable=0    failed=1   
```

File: playbooks/common/openshift-cluster/upgrades/v3_5/validator.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+###############################################################################
+# Post upgrade - Upgrade job storage
```
Comment (tbielawa) [inline playbooks/common/openshift-cluster/upgrades/v3_5/validator.yml]: This file seems to be called with the tag `pre_upgrade` in the above file (`upgrades/v3_5/upgrade_control_plane.yml`). However the comment header in this says it is for **Post upgrade - Upgrade job storage**.

But.... all it does is include `lib_openshift` and then *Check for invalid namespaces and SDN errors*.

This does not all match up. This **is** a pre-upgrade step, is it just mis-labeled?

Comment (sdodson) [inline playbooks/common/openshift-cluster/upgrades/v3_5/validator.yml]: copy/pasta, updated

File: roles/lib_openshift/library/oc_objectvalidator.py [hunk 2]
```diff
@@ -1380,7 +1380,14 @@ def _is_invalid_namespace(namespace):
                 all_invalid[invalid_msg] = invalid
 
         if failed:
-            return {'failed': True, 'msg': 'All objects are not valid.', 'state': 'list', 'results': all_invalid}
+            return {
+                'failed': True,
+                'msg': "All objects are not valid.  If you are a supported customer please contact \
+                        Red Hat Support providing the complete output above. If you are not a customer \
+                        please contact users@lists.openshift.redhat.com for assistance.",
```
Comment (sdodson) [inline roles/lib_openshift/library/oc_objectvalidator.py]: @tbielawa how do i do string continuations and have it not format the junk?

Comment (tbielawa) [inline roles/lib_openshift/library/oc_objectvalidator.py]: maybe like dis

https://github.com/openshift/openshift-ansible/pull/3585/files/4df830ca0624da671f548c7ab463eba42c58972e#diff-2ddfb0d4337f80bc7ae1662b2383a43bR40

Comment (tbielawa) [inline roles/lib_openshift/library/oc_objectvalidator.py]: Just kidding. That's python. Derp.

Comment (sdodson) [inline roles/lib_openshift/library/oc_objectvalidator.py]: @ashcrow pro-tips from a python masta on how to handle long lines without introducing formatting?

Comment (ashcrow) [inline roles/lib_openshift/library/oc_objectvalidator.py]: I'd recommend using ```()```'s.

Example

```python
                   'msg': (
                       "All objects are not valid.  If you are a supported customer please contact "
                       "Red Hat Support providing the complete output above. If you are not a customer "
                       "please contact users@lists.openshift.redhat.com for assistance."),
                   ...
```

Comment (sdodson) [inline roles/lib_openshift/library/oc_objectvalidator.py]: That's definitely readable but it still seems to get formatted by ansible. Perhaps ansible is doing something with the string. Anyway, it's a failure condition, we can deal with slightly imperfect formatting. Thanks.

Comment (ashcrow) [inline roles/lib_openshift/library/oc_objectvalidator.py]: @sdodson Ugh. Sorry.

Comment (sdodson) [inline roles/lib_openshift/library/oc_objectvalidator.py]: Nevermind, generated library versus src problems. *sigh*

**Comment by ashcrow** [comment]
Reviewing now.

**Comment by ashcrow** [comment]
aos-ci-test

File: roles/lib_openshift/library/oc_objectvalidator.py [hunk 3]
```diff
@@ -1380,7 +1380,16 @@ def _is_invalid_namespace(namespace):
                 all_invalid[invalid_msg] = invalid
 
         if failed:
-            return {'failed': True, 'msg': 'All objects are not valid.', 'state': 'list', 'results': all_invalid}
+            return {
```
Comment (ashcrow) [inline roles/lib_openshift/library/oc_objectvalidator.py]: Thank you for breaking this into a multi-line dict.

File: roles/lib_openshift/src/class/oc_objectvalidator.py [hunk 4]
```diff
@@ -72,6 +72,15 @@ def _is_invalid_namespace(namespace):
...
 
         if failed:
-            return {'failed': True, 'msg': 'All objects are not valid.', 'state': 'list', 'results': all_invalid}
+            return {
+                'failed': True,
+                'msg': (
+                    "All objects are not valid.  If you are a supported customer please contact "
+                    "Red Hat Support providing the complete output above. If you are not a customer "
+                    "please contact users@lists.openshift.redhat.com for assistance."
+                    ),
```
Comment (ashcrow) [inline roles/lib_openshift/src/class/oc_objectvalidator.py]: I'm not sure if this ending ```()``` location will cause a ``flake8`` or ``pylint`` error. But the change looks good.

File: roles/lib_openshift/library/oc_objectvalidator.py (hunk 3 above)
Comment (ashcrow) [inline roles/lib_openshift/library/oc_objectvalidator.py]: Thank you for making this a multi-line dict.

**Comment by ashcrow** [comment]
@sdodson something like this should work:

```diff
--- a/roles/lib_openshift/src/test/unit/test_oc_objectvalidator.py
+++ b/roles/lib_openshift/src/test/unit/test_oc_objectvalidator.py
@@ -891,7 +891,7 @@ class OCObjectValidatorTest(unittest.TestCase):
 
         # Assert
         self.assertTrue(results['failed'])
-        self.assertEqual(results['msg'], 'All objects are not valid.')
+        self.assertIn('All objects are not valid.', results['msg'])
```

**Comment by sdodson** [comment]
@ashcrow test cases updated

**Comment by sdodson** [comment]
Merged on travis being green and positive review, only change since last ci run was unit tests.

**Pull request merged**